### PR TITLE
disable conversions for empty fields

### LINKED ISSUE #9200: Convert an empty field from Latex to Unicode throws an exception
### JabRef version

Latest development branch build (please note build date below)

### Operating system

Windows

### Details on version and operating system

_No response_

### Checked with the latest development build

- [X] I made a backup of my libraries before testing the latest development version.
- [X] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

1. Select an entry by double-clicking.
2. Right-click on an empty field from the entry editor.
3. Select *Convert* then *LaTeX to Unicode*.

An exception is also raised when using the other conversion options *(HTML to LaTeX...)*.

### Appendix

<details>

<summary>Log File</summary>

```
java.lang.NullPointerException
	at java.base/java.util.Objects.requireNonNull(Objects.java:208)
	at org.jabref@100.0.0/org.jabref.model.strings.LatexToUnicodeAdapter.format(LatexToUnicodeAdapter.java:29)
	at org.jabref@100.0.0/org.jabref.logic.layout.format.LatexToUnicodeFormatter.format(LatexToUnicodeFormatter.java:26)
	at org.jabref@100.0.0/org.jabref.gui.fieldeditors.contextmenu.DefaultMenu.lambda$getConversionMenu$1(DefaultMenu.java:66)
	at javafx.base@19/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:86)
	at javafx.base@19/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:234)
	at javafx.base@19/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at javafx.base@19/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at javafx.base@19/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base@19/com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at javafx.base@19/com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:49)
	at javafx.base@19/javafx.event.Event.fireEvent(Event.java:198)
	at javafx.controls@19/javafx.scene.control.MenuItem.fire(MenuItem.java:459)
	at javafx.controls@19/com.sun.javafx.scene.control.ContextMenuContent$MenuItemContainer.lambda$createNodeMenuItemChildren$14(ContextMenuContent.java:1407)
	at javafx.base@19/com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(CompositeEventHandler.java:247)
	at javafx.base@19/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:80)
	at javafx.base@19/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:234)
	at javafx.base@19/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at javafx.base@19/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
	at javafx.base@19/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at javafx.base@19/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base@19/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base@19/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base@19/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base@19/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base@19/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base@19/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base@19/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base@19/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base@19/com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at javafx.base@19/com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
	at javafx.base@19/javafx.event.Event.fireEvent(Event.java:198)
	at javafx.graphics@19/javafx.scene.Scene$ClickGenerator.postProcess(Scene.java:3599)
	at javafx.graphics@19/javafx.scene.Scene$MouseHandler.process(Scene.java:3903)
	at javafx.graphics@19/javafx.scene.Scene.processMouseEvent(Scene.java:1887)
	at javafx.graphics@19/javafx.scene.Scene$ScenePeerListener.mouseEvent(Scene.java:2620)
	at javafx.graphics@19/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:411)
	at javafx.graphics@19/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:301)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at javafx.graphics@19/com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleMouseEvent$2(GlassViewEventHandler.java:450)
	at javafx.graphics@19/com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(QuantumToolkit.java:424)
	at javafx.graphics@19/com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(GlassViewEventHandler.java:449)
	at javafx.graphics@19/com.sun.glass.ui.View.handleMouseEvent(View.java:551)
	at javafx.graphics@19/com.sun.glass.ui.View.notifyMouse(View.java:937)
	at javafx.graphics@19/com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
	at javafx.graphics@19/com.sun.glass.ui.win.WinApplication.lambda$runLoop$3(WinApplication.java:184)
	at java.base/java.lang.Thread.run(Thread.java:833)

```

</details>

### PR DESCRIPTION
**Pull request by TheGor1lla**
Converting fields that were empty led to a Nullpointer Exception. This PR checks for null strings and replaces the return value with a blank String.

Fixes #9200 

- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [x] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [x] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/logic/formatter/bibtexfields/HtmlToLatexFormatter.java [hunk 1]
```diff
@@ -26,7 +26,7 @@ public class HtmlToLatexFormatter extends Formatter implements LayoutFormatter {
 
     @Override
     public String format(String text) {
-        String result = Objects.requireNonNull(text);
+        String result = Objects.requireNonNullElse(text, "");
```
Comment (HoussemNasri) [inline src/main/java/org/jabref/logic/formatter/bibtexfields/HtmlToLatexFormatter.java:29]: Although this solves the problem, I'm not a fan. I believe it is the client's responsibility to use valid arguments when calling the format method. If it does not, there may be a larger issue in the app that is causing this behavior. 
I believe you can apply the fix here and it will fix them all.

https://github.com/JabRef/jabref/blob/0f3b01696a610f4b0b319d6253c7f71ed35843c8/src/main/java/org/jabref/gui/fieldeditors/contextmenu/DefaultMenu.java#L58-L68.

File: src/test/java/org/jabref/logic/layout/format/LatexToUnicodeFormatterTest.java [hunk 2]
```diff
@@ -110,6 +110,11 @@ void keepUnknownCommandWithoutArgument() {
         assertEquals("\\aaaa", formatter.format("\\aaaa"));
     }
 
+    @Test
+    void keepUnknownCommandWithNullArgument() {
+        assertEquals("", formatter.format(null));
```
Comment (HoussemNasri) [inline src/test/java/org/jabref/logic/layout/format/LatexToUnicodeFormatterTest.java:115]: Good job on adding the test case, but since `null` isn't a valid argument for `format` I would assert a NullPointerException is thrown.

**Review by HoussemNasri — CHANGES_REQUESTED** [review]

File: src/main/java/org/jabref/logic/formatter/bibtexfields/HtmlToLatexFormatter.java (hunk 1 above)
Comment (TheGor1lla) [reply src/main/java/org/jabref/logic/formatter/bibtexfields/HtmlToLatexFormatter.java:29]: Thanks for the feedback :)
Thinking about it, would it be better to disable the menu items, if there's no text present?

Comment (HoussemNasri) [reply src/main/java/org/jabref/logic/formatter/bibtexfields/HtmlToLatexFormatter.java:29]: Nice idea! Preventing the exception is already an improvement and it should be enough to get the PR accepted but if you want to take it to the next level and disable menu items that's up to you.

Comment (TheGor1lla) [reply src/main/java/org/jabref/logic/formatter/bibtexfields/HtmlToLatexFormatter.java]: Great to hear, I went for that solution. The tests are excluded as I don't think they are necessary when the function isn't available to the user anymore.

File: src/main/java/org/jabref/gui/fieldeditors/contextmenu/DefaultMenu.java [hunk 3]
```diff
@@ -62,6 +62,7 @@ private static Menu getConversionMenu(TextInputControl textInputControl) {
             CustomMenuItem menuItem = new CustomMenuItem(new Label(converter.getName()));
             Tooltip toolTip = new Tooltip(converter.getDescription());
             Tooltip.install(menuItem.getContent(), toolTip);
+            menuItem.setDisable(textInputControl.textProperty().get() == null);
```
Comment (HoussemNasri) [inline src/main/java/org/jabref/gui/fieldeditors/contextmenu/DefaultMenu.java:65]: This will only disable the menu items at initialization. For instance, if you type something into the text field and delete it, the menu items aren't disabled. You need to call `menuItem.setDisable(textInputControl.textProperty().get() == null);` whenever textInputProperty changes. There is an elegant solution for this using `EasyBind#subscribe`. Also, you need to handle the case when the text is empty. We already have a method for that; see `StringUtil#isNullOrEmpty`.

**Review by HoussemNasri — APPROVED** [review]
LGTM! Looking forward to more PRs from you.
